router/pcap.go

```diff
@@ -1,19 +1,39 @@
 package router
 
 import (
-	"fmt"
-
 	"code.google.com/p/gopacket/pcap"
+	"fmt"
+	"net"
+	"sync"
 )
 
-type PcapIO struct {
-	handle *pcap.Handle
+type Pcap struct {
+	iface *net.Interface
+	bufSz int
+
+	// The libpcap handle for writing packets. It's possible thata
+	// single handle could be used for reading and writing, but
+	// we'd have to examine the performance implications.
+	writeHandle *pcap.Handle
+
+	// pcap handles are single-threaded, so we need to lock around
+	// uses of writeHandle.
+	writeHandleMutex sync.Mutex
```

Comment:
In the commit in which that change is introduced (69318506975), they correspond to the two uses of `router.IntraHost.InjectPacket` in `router.go`:  One is in the packet capturing goroutine, the other is in the udp listener goroutine.  This structure is as before, but before the packet capturing goroutine used the "reading" pcap handle to send, whereas now both cases use the "writing" handle, to avoid cluttering the `IntraHost` interface.

There are two corresponding cases after the introduction of `FlowOps`:

* `readUDP` -> `handleFrame` -> `sendToConsumer` -> `handleForwardedPacket` -> (via `injectFop`) `Pcap Send`
* `sniff` -> (via `fop` based on router's decision to forward over UDP) `curriedForward Send` -> `sendToConsumer` -> ditto (but in a different goroutine)